packages/core/src/store/abstract-store.ts

```diff
@@ -0,0 +1,63 @@
+import { StoreWrapperInterface } from './store-wrapper-interface.js'
+import { StoreInterface } from './store-interface.js'
+
+export abstract class AbstractStore<O, V> implements StoreInterface<O, V> {
```

Comment:
```suggestion
export abstract class AbstractStore<TKey, TValue> implements StoreInterface<TKey, TValue> {
```

Reply:
@ukstv , `O` here is not the type of the key (this is generally `string`), but rather the type of object from which a key is derived (`StreamID` for `StateStore`, `CommitID` for `AnchorRequestStore`), and then classes which extend this class need to implement methods for deriving a key from the object.

Perhaps this behaviour is misleading (as the object of type `O` is not stored anywhere, it's just used to derive the key)? Should I remove this generalisation altogether and just have a generic value type (assuming that the key is `string`)? Or should I rename `O` to `TObject`?

Reply:
Would vote for `TObject` or even `TKeyObject` (weak opinion, feel free to disregard the suggestion). To me semantically, it is still an advanced key :) But, again, that's a weak opinion, feel free to disregard.